shell/src/main/java/org/apache/accumulo/shell/Shell.java

```diff
@@ -584,40 +586,39 @@ public int start() throws IOException {
 
         // If tab completion is true we need to reset
         if (tabCompletion) {
-          if (userCompletor != null) {
-            reader.removeCompleter(userCompletor);
-          }
-
           userCompletor = setupCompletion();
-          reader.addCompleter(userCompletor);
+          ((LineReaderImpl) reader).setCompleter(userCompletor);
         }
 
-        reader.setPrompt(getDefaultPrompt());
-        input = reader.readLine();
+        input = reader.readLine(getDefaultPrompt());
         if (input == null) {
-          reader.println();
           return exitCode;
         } // User Canceled (Ctrl+D)
 
         execCommand(input, disableAuthTimeout, false);
       } catch (UserInterruptException uie) {
         // User Cancelled (Ctrl+C)
-        reader.println();
+        writer.println();
 
         String partialLine = uie.getPartialLine();
         if (partialLine == null || "".equals(uie.getPartialLine().trim())) {
           // No content, actually exit
           return exitCode;
         }
       } finally {
-        reader.flush();
+        writer.flush();
       }
     }
   }
 
   public void shutdown() {
     if (reader != null) {
-      reader.close();
+      try {
+        terminal.close();
+        reader = null;
+      } catch (IOException e) {
+        e.printStackTrace();
```

Comment:
I suggest adding a log.warn or log.error here to capture the exception within the logs.

Reply:
I concur. We don't generally just print stack traces with `e.printStackTrace()`.

Reply:
I will add.

Reply:
I noticed a printExcetion function included in the shell class already so I used that for this. Let me know if that suffices